Description: Better config default
 The default of Neutron make it so that it's not usable by default, and hard to
 parse with maintainer scripts. This patch fixes that.
Author: Thomas Goirand <zigo@debian.org>
Forwarded: no
Last-Update: 2013-11-27

Index: neutron/etc/l3_agent.ini
===================================================================
--- neutron.orig/etc/l3_agent.ini	2013-12-16 15:39:15.000000000 +0800
+++ neutron/etc/l3_agent.ini	2013-12-16 16:58:39.000000000 +0800
@@ -8,7 +8,7 @@
 
 # Example of interface_driver option for OVS based plugins (OVS, Ryu, NEC)
 # that supports L3 agent
-# interface_driver = neutron.agent.linux.interface.OVSInterfaceDriver
+interface_driver = neutron.agent.linux.interface.OVSInterfaceDriver
 
 # Use veth for an OVS interface or not.
 # Support kernels with limited namespace support
Index: neutron/etc/neutron.conf
===================================================================
--- neutron.orig/etc/neutron.conf	2013-12-16 16:57:07.000000000 +0800
+++ neutron/etc/neutron.conf	2013-12-16 16:58:39.000000000 +0800
@@ -45,7 +45,7 @@
 # api_extensions_path =
 
 # Neutron plugin provider module
-# core_plugin =
+core_plugin = neutron.plugins.openvswitch.ovs_neutron_plugin.OVSNeutronPluginV2
 
 # Advanced service modules
 # service_plugins =
@@ -119,16 +119,16 @@
 # SSL certification authority file (valid only if SSL enabled)'
 # kombu_ssl_ca_certs =
 # IP address of the RabbitMQ installation
-# rabbit_host = localhost
+rabbit_host = localhost
 # Password of the RabbitMQ server
-# rabbit_password = guest
+rabbit_password = guest
 # Port where RabbitMQ server is running/listening
 # rabbit_port = 5672
 # RabbitMQ single or HA cluster (host:port pairs i.e: host1:5672, host2:5672)
 # rabbit_hosts is defaulted to '$rabbit_host:$rabbit_port'
 # rabbit_hosts = localhost:5672
 # User ID used for RabbitMQ connections
-# rabbit_userid = guest
+rabbit_userid = guest
 # Location of a virtual RabbitMQ installation.
 # rabbit_virtual_host = /
 # Maximum retries with trying to connect to RabbitMQ
@@ -299,7 +299,7 @@
 # Use "sudo neutron-rootwrap /etc/neutron/rootwrap.conf" to use the real
 # root filter facility.
 # Change to "sudo" to skip the filtering and just run the comand directly
-# root_helper = sudo
+root_helper = sudo neutron-rootwrap /etc/neutron/rootwrap.conf
 
 # =========== items for agent management extension =============
 # seconds between nodes reporting state to server; should be less than
@@ -319,11 +319,10 @@
 
 [database]
 # This line MUST be changed to actually run the plugin.
-# Example:
-# connection = mysql://root:pass@127.0.0.1:3306/neutron
+# Example: connection = mysql://root:pass@127.0.0.1:3306/neutron
 # Replace 127.0.0.1 above with the IP address of the database used by the
 # main neutron server. (Leave it as is if the database runs on this host.)
-# connection = sqlite://
+connection = sqlite:///var/lib/neutron/neutrondb
 
 # The SQLAlchemy connection string used to connect to the slave database
 # slave_connection =
Index: neutron/etc/lbaas_agent.ini
===================================================================
--- neutron.orig/etc/lbaas_agent.ini	2013-12-10 00:11:29.000000000 +0800
+++ neutron/etc/lbaas_agent.ini	2013-12-16 16:58:39.000000000 +0800
@@ -13,7 +13,7 @@
 
 # Example of interface_driver option for OVS based plugins (OVS, Ryu, NEC, NVP,
 # BigSwitch/Floodlight)
-# interface_driver = neutron.agent.linux.interface.OVSInterfaceDriver
+interface_driver = neutron.agent.linux.interface.OVSInterfaceDriver
 
 # Use veth for an OVS interface or not.
 # Support kernels with limited namespace support
Index: neutron/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini
===================================================================
--- neutron.orig/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini	2013-12-16 15:39:15.000000000 +0800
+++ neutron/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini	2013-12-16 16:58:39.000000000 +0800
@@ -30,7 +30,7 @@
 #          point setting tunnel_type below will be required to enable
 #          tunneling.
 #
-# enable_tunneling = False
+enable_tunneling = False
 
 # (StrOpt) The type of tunnel network, if any, supported by the plugin. If
 # this is set, it will cause tunneling to be enabled. If this is not set and
@@ -44,8 +44,8 @@
 # enumerating ranges of GRE or VXLAN tunnel IDs that are available for
 # tenant network allocation if tenant_network_type is 'gre' or 'vxlan'.
 #
-# tunnel_id_ranges =
 # Example: tunnel_id_ranges = 1:1000
+tunnel_id_ranges = 1:1000
 
 # Do not change this parameter unless you have a good reason to.
 # This is the name of the OVS integration bridge. There is one per hypervisor.
@@ -70,7 +70,7 @@
 # empty for the server. Set local-ip to be the local IP address of
 # this hypervisor.
 #
-# local_ip =
+local_ip = 127.0.0.1
 
 # (ListOpt) Comma-separated list of <physical_network>:<bridge> tuples
 # mapping physical network names to the agent's node-specific OVS
@@ -123,30 +123,5 @@
 
 [securitygroup]
 # Firewall driver for realizing neutron security group function.
-# firewall_driver = neutron.agent.firewall.NoopFirewallDriver
-# Example: firewall_driver = neutron.agent.linux.iptables_firewall.OVSHybridIptablesFirewallDriver
-
-#-----------------------------------------------------------------------------
-# Sample Configurations.
-#-----------------------------------------------------------------------------
-#
-# 1. With VLANs on eth1.
-# [database]
-# connection = mysql://root:nova@127.0.0.1:3306/ovs_neutron
-# [OVS]
-# network_vlan_ranges = default:2000:3999
-# tunnel_id_ranges =
-# integration_bridge = br-int
-# bridge_mappings = default:br-eth1
-# [AGENT]
-# Add the following setting, if you want to log to a file
-#
-# 2. With tunneling.
-# [database]
-# connection = mysql://root:nova@127.0.0.1:3306/ovs_neutron
-# [OVS]
-# network_vlan_ranges =
-# tunnel_id_ranges = 1:1000
-# integration_bridge = br-int
-# tunnel_bridge = br-tun
-# local_ip = 10.0.0.3
+# Example: firewall_driver = neutron.agent.firewall.NoopFirewallDriver
+firewall_driver = neutron.agent.linux.iptables_firewall.OVSHybridIptablesFirewallDriver
